nixos/modules/services/hardware/thermald.nix

```diff
@@ -4,27 +4,392 @@ with lib;
+          next.
+        '';
+      };
+
+      CoolingDevices = mkOption {
+        type = (listOf tripPointCoolingDeviceConfig);
+        default = [];
+        description = ''
+          The cooling devices to use for this trip point
+        '';
+      };
+    };
+  };
+
+  tripPointCoolingDeviceConfig = with types; submodule {
+    options = {
+      Index = mkOption {
+        type = nullOr int;
+        default = null;
+        description = "The index of this cooling device";
+      };
+
+      Type = mkOption {
+        type = string;
+        description = ''
+          The type of cooling device.
+        '';
+      };
+
+      Influence = mkOption {
+        type = nullOr int;
+        default = null;
+        description = ''
+          Influence will be used order cooling devices. First cooling device will be
+          used, which has highest influence.
+        '';
+      };
+
+      SamplingPeriod = mkOption {
+        type = nullOr int;
+        default = null;
+        description = ''
+          Delay in using this cdev, this takes some time to actually cool a zone
+        '';
+      };
+
+      TargetState = mkOption {
+        type = nullOr int;
+        default = null;
+        description = ''
+          Set a specific state of this cooling device when this trip is violated
+        '';
+      };
+    };
+  };
+
+  coolingDeviceConfig = with types; submodule {
+    options = {
+      Type = mkOption {
+        type = string;
+        description = "The type of this cooling device";
+      };
+
+      MinState = mkOption {
+        type = nullOr int;
+        default = null;
+        description = "The min state of this cooling device";
+      };
+
+      IncDecStep = mkOption {
+        type = nullOr int;
+        default = null;
+        description = "The IncDecStep";
+      };
+
+      ReadBack = mkOption {
+        type = nullOr int;
+        default = null;
+        description = "The ReadBack";
+      };
+
+      MaxState = mkOption {
+        type = nullOr int;
+        default = null;
+        description = "The MaxState";
+      };
+
+      DebouncePeriod = mkOption {
+        type = nullOr int;
+        default = null;
+        description = "The DebouncePeriod";
+      };
+
+      PidControl = mkOption {
+        description = ''
+          If there are no PID parameters, compensation increase step wise and exponentially (if
+          single step is not able to change trend). Alternatively a PID parameters can be
+          specified then next step  will use PID calculation using provided PID constants.
+        '';
+        default = {};
+        type = submodule {
+          options = {
+            kp = mkOption {
+              type = string;
+              description = "kp";
+            };
+
+            kd = mkOption {
+              type = string;
+              description = "kd";
+            };
+
+            ki = mkOption {
+              type = string;
+              description = "ki";
```

Comment:
Also a lot of options could use better descriptions, e.g. these ones. But it might be pretty hard to find out what these exactly represent.

Reply:
I'd love to give these better descriptions but I have no idea what they do. The documentation is pretty sparse on them and I'm assuming if you want to set them then you know what they are :eyes:

With that said if anyone knows what they do I'm happy to include a better description!